src/package-resolver.js

```diff
@@ -352,7 +353,9 @@ export default class PackageResolver {
     this.patterns[pattern] = info;
 
     const byName = (this.patternsByPackage[info.name] = this.patternsByPackage[info.name] || []);
-    byName.push(pattern);
+    if (byName.indexOf(pattern) === -1) {
```

Comment:
I feel like it would be a lot more efficient to use a `Set` here since the ordering doesn't seem to be important?

I guess that'd be a refactor we need to do in a follow up?

Reply:
In that specific function yes, `Set` would be better, but the rest of the functions use `Array.map` and `Array.filter` on that object, which are not available on `Set` so we would have to convert Set to Array all over the place: `[...patterns].map(` which is probably worse overall for memory and performance usage. (expanding the set to an array frequently, vs infrequent `.indexOf` calls on small arrays)